src/main/java/org/example/QuantileCalculator.java

```diff
@@ -0,0 +1,65 @@
+package org.example;
+
+/**
+ * Provides methods to calculate the probability density function (PDF) and cumulative distribution function (CDF)
+ * of a standard normal distribution, and to find quantiles using the Newton-Raphson method.
+ */
+public class QuantileCalculator {
+    private static final double EPSILON = 1e-10;
+    private static final double SQRT_TWO_PI = Math.sqrt(2 * Math.PI);
+
+    /**
+     * Calculates the value of the probability density function (PDF) for a standard normal distribution at a given x.
+     *
+     * @param x The point at which to evaluate the PDF.
+     * @return The PDF value at x.
+     */
+    public static double normalPdf(double x) {
+        return Math.exp(-0.5 * x * x) / SQRT_TWO_PI;
+    }
+
+    /**
+     * Calculates the cumulative distribution function (CDF) for a standard normal distribution up to a given x using Simpson's rule.
+     *
+     * @param x The upper limit to which the CDF is calculated.
+     * @return The CDF value up to x.
+     */
+    public static double normalCdf(double x) {
+        final int N = 10000;  // More segments mean more accuracy
```

Comment:
Insted of using more segments to get accuracy, I suggest you use Gauss-Legendre quadrature with fixed number of nodes. Verify which number of nodes gives satisfactory accuracy.